### ISSUE TITLE
Testing Linear Regression Assumptions

### ISSUE THREAD

**Issue by NicoGalli** [feature-engine/feature_engine#581]
Checking model assumptions is like commenting code. Everybody should be doing it often, but it sometimes ends up being overlooked in reality. A failure to do either can result in a lot of time being confused, going down rabbit holes, and can have pretty serious consequences from the model not being interpreted correctly.

What about creating the following functions to test linear regression assumptions?

    linear_assumption()
    normal_errors_assumption()
    multicollinearity_assumption()
    autocorrelation_assumption()
    homoscedasticity_assumption()

Please check this amazing post: https://jeffmacaluso.github.io/post/LinearRegressionAssumptions/

**Comment by Morgan-Sell** [comment]
Nice idea, @NicoGalli!

Has anyone started this task? If not, I'll take it on.

**Comment by NicoGalli** [comment]
Hi @Morgan-Sell!

Yes, please. Dont hesitate to assign this task to you and start working on it.

**Comment by Morgan-Sell** [comment]
Hi @NicoGalli,

I understand the request. However, I'm questioning whether this task should be within the scope of feature-engine. feature-engine focuses on automating feature engineering/selection that is compatible with scikit-learn.

Assessing linear regression assumptions falls under model evaluation, which I believe to beyond the scope of feature-engine.

@solegalli, do you have any thoughts?

**Comment by solegalli** [comment]
Yes, I think that for the time being, this will be out of scope. There are other issues with higher priority.

Maybe in a few years :p
